src/main/java/no/nav/veilarbarena/config/ApplicationConfig.java

```diff
@@ -122,11 +126,24 @@ public ArenaOrdsTokenProviderClient arenaOrdsTokenProvider() {
         return new ArenaOrdsTokenProviderClient(createArenaOrdsUrl());
     }
 
-    @Bean
     public ArenaOrdsClient arenaOrdsClient(ArenaOrdsTokenProviderClient arenaOrdsTokenProviderClient) {
         return new ArenaOrdsClientImpl(createArenaOrdsUrl(), arenaOrdsTokenProviderClient::getToken);
     }
 
+    @Bean
+    public PoaoTilgangClient poaoTilgangClient(EnvironmentProperties properties, AzureAdMachineToMachineTokenClient tokenClient) {
+        return new PoaoTilgangCachedClient(
+                new PoaoTilgangHttpClient(
+                        properties.getPoaoTilgangUrl(),
+                        () -> tokenClient.createMachineToMachineToken(properties.getPoaoTilgangScope()),
+                        RestClient.baseClient()
+                ),
+                null,
```

Comment:
Du får ikke nullpointer her altså? Kan alikevel vurdere å legge inn:
```
private final Cache<PolicyInput, Decision> policyInputToDecisionCache = Caffeine.newBuilder()
			.expireAfterWrite(Duration.ofMinutes(30))
			.build();
	private final Cache<UUID, List<AdGruppe>> navAnsattIdToAzureAdGrupperCache = Caffeine.newBuilder()
			.expireAfterWrite(Duration.ofMinutes(30))
			.build();
	private final Cache<String, Boolean> norskIdentToErSkjermetCache = Caffeine.newBuilder()
			.expireAfterWrite(Duration.ofMinutes(30))
			.build();
```
i toppen og så legge de inn i selve funksjonen for å få cachene:
```
 @Bean
    public PoaoTilgangClient poaoTilgangClient(EnvironmentProperties properties, AzureAdMachineToMachineTokenClient tokenClient) {
        return new PoaoTilgangCachedClient(
                new PoaoTilgangHttpClient(
                        properties.getPoaoTilgangUrl(),
                        () -> tokenClient.createMachineToMachineToken(properties.getPoaoTilgangScope()),
                        RestClient.baseClient()
                ),
                policyInputToDecisionCache,
                navAnsattIdToAzureAdGrupperCache,
                norskIdentToErSkjermetCache
        );
    }
```

Reply:
Har tatt det inn